functions/supported.pp

```diff
@@ -0,0 +1,19 @@
+# @summary Returns whether the currently loaded OS is supported by the module
+#
+# This function uses the current facts to check if the current Operating System
+# and its release or major release is supported.
+#
+# @param mod
+#  The module name to check.
+#
+# @example Using the Puppet built in global $module_name
+#   openvmtools::supported($module_name)
+function openvmtools::supported(String[1] $mod) >> Boolean {
+  $metadata = load_module_metadata($mod)
+  unless $metadata {
+    fail("Unable to load metadata for mod '${mod}'")
+  }
```

Comment:
This is probably already redundant:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/09a9b975ca4da2735cbba1cf2417d9ce2c174ca2/lib/puppet/parser/functions/load_module_metadata.rb